lib/generate_letters.ml

```diff
@@ -61,6 +61,19 @@ let realistic_random_letter () =
   if Random.int 26 < 5 then get_random_letter vowel_amounts
   else get_random_letter consonant_amounts
 
+let rec has_vowel = function
+  | [] -> false
+  | h :: t ->
+      if List.mem h [ 'A'; 'E'; 'I'; 'O'; 'U' ] then true
+      else has_vowel t
+
+let insert_vowel lst =
+  match lst with
+  | [] -> []
+  | _ :: t -> get_random_letter vowel_amounts :: t
```

Comment:
does `insert_vowel` always change the first letter to be a vowel? if so, why does it need a match statement? also, please add a comment explaining how this function works

Reply:
> does `insert_vowel` always change the first letter to be a vowel? if so, why does it need a match statement? also, please add a comment explaining how this function works

Looks like it does only change the first letter to be a vowel. I can change it to a map if that's better?

Reply:
it doesn't really matter, i was just wondering why ... you can also leave it as is but please add a comment

Reply:
Comments added.

Reply:
yea it does i didnt think it mattered where the vowel was